### PR TITLE
FEATURE: SD-869 Geo chart displays friendly message when Mapbox token is missed

### PR DESCRIPTION
**Pull request by dlcd2203**
---

[Check PR owner responsibilities](https://confluence.intgdc.com/display/Development/Code-reviews#Code-reviews-Ownerresponsibilities)

Supported PR commands:

Command | Description
--- | ---
`extended test - examples` | Live examples tests
`extended test - storybook` | Storybook screenshot tests

See more [options](https://confluence.intgdc.com/display/kbhr/How+to+work+with+git+and+Github#HowtoworkwithgitandGithub-Extendedchecks).

# PR Checklist

- [x] Verify code changes ([Checklist](https://confluence.intgdc.com/display/Development/Code-reviews+checklist), [Best practices](https://confluence.intgdc.com/display/Development/Code-reviews+best+practices))
- [x] [Verify pull-request formalities](https://confluence.intgdc.com/display/Development/Code-reviews)
- [x] Change was tested by using [gdc-dev-release](https://confluence.intgdc.com/display/~tomas.vojtasek/Private+NPM) in [gdc-analytical-designer](https://github.com/gooddata/gdc-analytical-designer) and [gdc-dashboards](https://github.com/gooddata/gdc-dashboards) (if applicable)
- [x] Migration guide (for major changes) is mentioned in [CHANGELOG.md](../blob/master/CHANGELOG.md).
- [x] Successful `extended test - examples`
- [x] Successful `extended test - storybook`
- [x] Checked yarn.lock consistency (no dep. duplicities especially Goodstrap)

# Related PRs

- gdc-analytical-designer: https://github.com/gooddata/gdc-analytical-designer/pull/2997
- gdc-dashboards: https://github.com/gooddata/gdc-dashboards/pull/2712

# Related Jira tasks

 SD-869: https://jira.intgdc.com/browse/SD-869

### CODE REVIEW

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/6930/ : FAILURE in 5m 43s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/2828/ : SUCCESS in 3s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/6931/ : SUCCESS in 3m 51s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/2829/ : SUCCESS in 3s

**Comment by dlcd2203** [comment]
extended test - storybook

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/3621/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/2830/ : SUCCESS in 3s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/2988/ : SUCCESS in 14m 37s

**Comment by dlcd2203** [comment]
extended test - examples

**Comment by yenkins** [comment]
:arrow_right: Started Live Examples deploy to:
https://gdc-live-examples-1526.herokuapp.com

It will be destroyed on merge/close. See [ci-infra](https://github.com/gooddata/ci-infra/blob/master/jenkins/jobs/templates/orchestrators/client-pr-dispatcher.groovy) and [gdc-ci.yaml](https://github.com/gooddata/gooddata-react-components/blob/master/gdc-ci.yaml).

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/2831/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/2989/ : SUCCESS in 24m 06s

File: src/components/core/base/test/GeoValidatorHOC.spec.tsx [hunk 1]
```diff
@@ -152,4 +140,41 @@ describe("GeoValidatorHOC", () => {
             },
         ]);
     });
+
+    it("should show GEO_MAPBOX_TOKEN_MISSING error", () => {
```
Comment (ptquang86) [inline src/components/core/base/test/GeoValidatorHOC.spec.tsx:144]: please move these new tests next to "should show GEO_LOCATION_MISSING error" test

File: src/translations/en-US.json [hunk 2]
```diff
@@ -194,6 +194,16 @@
         "comment": "",
         "limit": 0
     },
+    "visualization.ErrorMessageMissingMapboxToken": {
```
Comment (ptquang86) [inline src/translations/en-US.json:197]: could we reuse the text `visualization.ErrorMessageGeneric`?

File: src/translations/en-US.json [hunk 3]
```diff
@@ -194,6 +194,16 @@
...
+        "value": "Sorry, we can't display this insight",
+        "comment": "",
+        "limit": 0
+    },
+    "visualization.ErrorDescriptionMissingMapboxToken": {
+        "value": "An API access token is required to use Mapbox GL in Geo chart (pushpins)",
+        "comment": "",
```
Comment (patrikbraborec) [inline src/translations/en-US.json:204]: Here should be information - Mapbox GL should not be translated. Please, fill it out. Thanks. :)

File: src/helpers/errorHandlers.ts [hunk 4]
```diff
@@ -53,6 +53,10 @@ export function generateErrorMap(intl: IntlShape): IErrorMap {
             message: intl.formatMessage({ id: "visualization.ErrorMessageGeneric" }),
             description: intl.formatMessage({ id: "visualization.ErrorDescriptionGeneric" }),
         },
+        [ErrorStates.GEO_MAPBOX_TOKEN_MISSING]: {
+            message: intl.formatMessage({ id: "visualization.ErrorMessageMissingMapboxToken" }),
```
Comment (naibinh) [inline src/helpers/errorHandlers.ts:57]: it should be replaced to `visualization.ErrorMessageGeneric`

File: src/components/core/base/test/GeoValidatorHOC.spec.tsx (hunk 1 above)
Comment (dlcd2203) [reply src/components/core/base/test/GeoValidatorHOC.spec.tsx]: fixed

File: src/translations/en-US.json (hunk 2 above)
Comment (dlcd2203) [reply src/translations/en-US.json]: fixed

File: src/translations/en-US.json (hunk 3 above)
Comment (dlcd2203) [reply src/translations/en-US.json]: added. Thanks :)

File: src/components/core/base/GeoValidatorHOC.tsx [hunk 5]
```diff
@@ -58,9 +60,10 @@ export function geoValidatorHOC<T>(InnerComponent: React.ComponentClass<T>): Rea
 
         private initError() {
             const {
-                config: { mdObject: { buckets = [] } = {} },
+                config: { mdObject: { buckets = [] } = {}, mapboxToken },
             } = this.props;
             this.isLocationMissing = isLocationMissing(buckets);
+            this.isMapboxTokenMissing = !!!mapboxToken;
```
[the remark this answers is not in the document]
Comment (dlcd2203) [reply src/components/core/base/GeoValidatorHOC.tsx]: fixed

File: src/helpers/errorHandlers.ts (hunk 4 above)
Comment (dlcd2203) [reply src/helpers/errorHandlers.ts]: fixed

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/6932/ : SUCCESS in 4m 16s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/2832/ : SUCCESS in 3s

**Comment by naibinh** [comment]
LGTM

**Comment by ptquang86** [comment]
LGTM

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/6959/ : SUCCESS in 11m 22s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/2873/ : SUCCESS in 5s

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/3638/ : SUCCESS in 17m 05s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/6961/ : SUCCESS in 3m 14s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-merge-compliance-check/1686/ : SUCCESS in 12s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-examples-zuul-docker/2447/ : SUCCESS in 25m 08s

**Pull request merged**